crates/comms/livekit_web_bindings.js

```diff
@@ -0,0 +1,180 @@
+            }).catch(error => {
+                console.error('Failed to enable microphone:', error);
+                currentMicTrack = null;
+            });
+        }
+    } else {
+        // Disable microphone
+        if (currentMicTrack) {
+            // Unpublish from all active rooms
+            const unpublishPromises = Array.from(activeRooms).map(room => 
+                room.localParticipant.unpublishTrack(currentMicTrack).catch(error => {
+                    console.error(`Failed to unpublish from room: ${error}`);
+                })
+            );
+            
+            Promise.all(unpublishPromises).then(() => {
+                currentMicTrack.stop();
+                currentMicTrack = null;
+                console.log('Microphone disabled successfully for all rooms');
+            }).catch(error => {
+                console.error('Failed to disable microphone:', error);
+            });
+        }
+    }
+}
+
+export function is_microphone_available() {
+    // Check if getUserMedia is available
+    const res = !!(navigator.mediaDevices && navigator.mediaDevices.getUserMedia)
+    return res;
+}
+
+export async function publish_data(room, data, reliable, destinations) {
+    const options = {
+        reliable: reliable,
+        destination: destinations.length > 0 ? destinations : undefined,
+    };
+    
+    await room.localParticipant.publishData(data, options);
+}
+
+export async function publish_audio_track(room, track) {
+    const publication = await room.localParticipant.publishTrack(track, {
+        source: LivekitClient.Track.Source.Microphone,
+    });
+    return publication.trackSid;
+}
+
+export async function unpublish_track(room, sid) {
+    const publication = room.localParticipant.trackPublications.get(sid);
+    if (publication) {
+        await room.localParticipant.unpublishTrack(publication.track);
+    }
+}
+
+export async function close_room(room) {
+    // Remove from active rooms set
+    activeRooms.delete(room);
+    
+    // If this was the last room and mic is active, clean up
+    if (activeRooms.size === 0 && currentMicTrack) {
+        currentMicTrack.stop();
+        currentMicTrack = null;
+    }
+    
+    await room.disconnect();
+}
+
+// Remove create_audio_track and send_audio_frame functions as they're no longer needed
+
+export function set_room_event_handler(room, handler) {
+    room.on(LivekitClient.RoomEvent.DataReceived, (payload, participant) => {
+        handler({
+            type: 'data_received',
+            payload: payload,
+            participant: participant,
+        });
+    });
+    
+    room.on(LivekitClient.RoomEvent.TrackSubscribed, (track, publication, participant) => {
+        // For audio tracks, automatically play them
+        if (track.kind === 'audio') {
+            const audioElement = track.attach();
+            audioElement.play().catch(e => console.warn('Failed to play audio:', e));
```

Comment:
this means we lose the spatial position for the received audio...

it seems a shame to send audio data back to rust and re-process it though ... we could possibly keep the processing in js and just send the volume / panning info from rust to js (if audioElement allows us to manage the left/right channel volumes separately).

this can be a follow up though, maybe to consider alongside threading the audio better for web - we may want all audio processing to be browser-side.

Reply:
yes that works. don't worry about setting the values yet, we can do that later.